### PR TITLE
Add module for video extraction

### PR DESCRIPTION
**Pull request by KomalBhadani27**
Added video_extractor module which:

- Validates whether a tag is video tag or not.
- If tag is video tag, extracts the tag and creates and returns the Video Content object.

### CODE REVIEW

File: extraction/video_extractor.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+def get_size(node):
+    """Returns the size of the video frame"""
+
+    if node.has_attr('width') and node.has_attr('height'):
+        return node['width'], node['height']
+    return 0, 0
+
+
+def is_video(node):
```
Comment (detel) [inline extraction/video_extractor.py:19]: implement IContentExtractor

Comment (detel) [inline extraction/video_extractor.py:19]: Add unit test for this method

File: extraction/video_extractor.py [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+creates and returns the Video object"""
+
+import re
+
+from extraction.data_models.video import Video
+
+VIDEO_EXTENSIONS_PATTERN = re.compile(r'^.*\.(mp4|ogg|mov|webm)')
+
+
+def get_size(node):
```
Comment (detel) [inline extraction/video_extractor.py:11]: Move private functions down. Also, add _ to it.

File: extraction/video_extractor.py [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            video_urls.append(node['src'])
+        elif node.contents:
+            for child in node.contents:
+                if child.name == 'source':
+                    video_urls.append(child['src'])
+        video = Video(video_urls)
+        video.width, video.height = get_size(node)
+        return True, video
+    else:
+        if node.name == 'embed' and node.has_attr('src'):
```
Comment (detel) [inline extraction/video_extractor.py:35]: merge this to elif

File: extraction/extractor.py [hunk 4]
```diff
@@ -84,8 +85,10 @@ def __extract_data_from_html_body(self, node):
     def __is_content(self, node):
         """This function will extract valid content and return it"""
 
-        # This function will be updated in next iteration
-        _ = node
         _ = self.soup
+        # checks if content is video
+        is_content, video_content = video_extractor.is_video(node)
+        if is_content:
+            return True, video_content
```
Comment (detel) [inline extraction/extractor.py:92]: Move video_Extractor to list. And loop through the lists.

File: extraction/video_extractor.py (hunk 1 above)
Comment (KomalBhadani27) [reply extraction/video_extractor.py]: Implemented!

File: extraction/video_extractor.py (hunk 2 above)
Comment (KomalBhadani27) [reply extraction/video_extractor.py]: moved to extraction_utils

File: extraction/video_extractor.py (hunk 3 above)
Comment (KomalBhadani27) [reply extraction/video_extractor.py]: Updated accordingly!

File: extraction/extractor.py (hunk 4 above)
Comment (KomalBhadani27) [reply extraction/extractor.py]: Changed is_video to validate_and_extract method of class VideoExtractor in video_extractor.

Comment (KomalBhadani27) [reply extraction/extractor.py]: Added tuple for IContentExtractor and updated accordingly.

File: extraction/video_extractor.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+from extraction.interface_content_extractor import IContentExtractor
+from extraction.utils import extraction_utils as e_utils
+
+VIDEO_EXTENSIONS_PATTERN = re.compile(r'^.*\.(mp4|ogg|mov|webm)')
+
+
+class VideoExtractor(IContentExtractor):
+    """This class inherits IContentExtractor for extracting video"""
+
+    def validate_and_extract(self, node: bs4.element):
```
Comment (detel) [inline extraction/video_extractor.py:18]: unit test?

File: extraction/video_extractor.py (hunk 1 above)
Comment (KomalBhadani27) [reply extraction/video_extractor.py]: Updated!

File: extraction/video_extractor.py (hunk 5 above)
Comment (KomalBhadani27) [reply extraction/video_extractor.py:18]: Added!

File: extraction/utils/extraction_utils.py [hunk 6]
```diff
@@ -0,0 +1,9 @@
+"""This script contains helper utilities for extractors"""
+
+
+def get_size(node):
```
Comment (droidkid) [inline extraction/utils/extraction_utils.py:4]: name to get_video_size. Probably name the file to video_utils

(Will this method be used in multiple files, you could just make it part of video_extractor.py - upto you)

Comment (droidkid) [reply extraction/utils/extraction_utils.py:4]: I just saw that the image PR also a get_size method - is it using the same method? Probably you could renamt to get_media_size and the file to media_extraction_utils.py

File: tests/test_extraction/test_video_extraction.py [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+
+
+def __assert_video(actual_video, expected_video):
+    """This is custom assert to compare actual and
+       expected video content"""
+
+    assert isinstance(actual_video, Video)
+    assert actual_video.video_urls == expected_video.video_urls
+    assert actual_video.width == expected_video.width
+    assert actual_video.height == expected_video.height
```
Comment (detel) [inline tests/test_extraction/test_video_extraction.py:56]: Private method at bottom

File: extraction/utils/extraction_utils.py (hunk 6 above)
Comment (KomalBhadani27) [reply extraction/utils/extraction_utils.py]: Added!

File: tests/test_extraction/test_video_extraction.py (hunk 7 above)
Comment (KomalBhadani27) [reply tests/test_extraction/test_video_extraction.py]: Updated accordingly!

**Pull request merged**